### PR TITLE
pkg/serverless/appsec: set the sampling priority upon appsec events

### PR DESCRIPTION
**Pull request by Julio-Guerra**
### What does this PR do?

Modify the trace-agent optional field `ModifySpan()` to now pass the `*pb.TraceChunk` the span is in and so that the callback function can set the sampling priority that is stored in the trace chunk struct.

Note that `normalizeChunk` could be used instead if we move it after the calls to `ModifySpan`, but I feared the it was required by the function calls in between its current location and the spans loop. 

### Motivation

Allow AppSec to set the sampling priority when security events are found.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/agent/agent.go [hunk 1]
```diff
@@ -68,8 +68,8 @@ type Agent struct {
 	// DiscardSpan will be called on all spans, if non-nil. If it returns true, the span will be deleted before processing.
 	DiscardSpan func(*pb.Span) bool
 
-	// ModifySpan will be called on all spans, if non-nil.
-	ModifySpan func(*pb.Span)
+	// ModifySpan will be called on all spans, if non-nil, along with the corresponding trace chunk.
```
Comment (ahmed-mez) [inline pkg/trace/agent/agent.go:71]: Just to make it as clear as possible for future developers who will use the API: Let's document that the modifier function is called in a loop and that the modifications introduced to `*pb.TraceChunk` in iteration N can be overwritten in iteration N+1!

**Review by ahmed-mez — APPROVED** [review]
One minor suggestion, LGTM otherwise!

**Comment by ahmed-mez** [comment]
Also please don't merge until the CI is fixed! 😅

File: pkg/trace/agent/agent.go (hunk 1 above)
Comment (Julio-Guerra) [reply pkg/trace/agent/agent.go]: Done in ed3bac00dd

**Review by maxday — APPROVED** [review]
LGTM! I just have one suggestion: should we rename the span modifier? It seems that now it modifies for than the span (ie the priority here, but could be use in other use cases)

**Pull request merged**